submissions/f3d0t/friends-app/app.js

```diff
@@ -0,0 +1,166 @@
+const MAIN = document.querySelector(".main");
+const FILTERS = document.querySelector(".filters");
+const RESET_BUTTON = document.querySelector(".filters__button");
+const DAY_NIGHT = document.querySelector(".day_night");
+const FRIENDS_COUNT = 60;
+const REQUEST_LINK = `https://randomuser.me/api/?results=${FRIENDS_COUNT}`;
+
+const fetchData = async (requestLink) => {
+	try {
+		const response = await fetch(requestLink);
+		const { results } = await response.json();
+		return results;
+	} catch (error) {
+		alert("HTTP-Error: " + error);
+	}
+};
+
+const initApp = (friendsDataArray) => {
+	const friends = new FriendsList(friendsDataArray, MAIN);
+	friends.renderCards();
+	bindEventListeners(friends);
+};
+
+const bindEventListeners = (friends) => {
+	FILTERS.addEventListener("input", ({ target }) => {
+		if (target.name == "search") {
+			friends.filterBySearch(target.value);
+		}
+		if (target.name == "gender") {
+			friends.filterByGender(target.id);
+			FILTERS.elements.search.value = "";
+			FILTERS.elements.name.forEach((radioButton) => {
+				if (radioButton.checked) {
+					friends.sortByName(radioButton.id);
+				}
+			});
+			FILTERS.elements.age.forEach((radioButton) => {
```

Comment:
If I understand correctly, you are using `NodeList#forEach` here in order to get "one" selected radio input, right?

If so, wrap them with `fieldset` and then just get checked input, because right now it is confusing.

Reply:
Ok, thanks=)